### PR TITLE
Public API / Doc cleanup

### PR DESCRIPTION
**Pull request by lu-zero**
Use `cargo doc --open` to see the difference.

Would probably make sense to group all the configuration structs in some way e.g. all the color definition structs in a `color` module.

### CODE REVIEW

File: benches/bench.rs [hunk 1]
```diff
@@ -12,18 +12,19 @@ mod transform;
...
+use rav1e::bench::partition::*;
+use rav1e::bench::predict::*;
+use rav1e::bench::rdo::*;
+
 use crate::transform::transform;
 
 use criterion::*;
 use std::time::Duration;
 
+
```
Comment (kodabb) [inline benches/bench.rs]: nit: extra linebreak

File: src/bin/common.rs [hunk 2]
```diff
@@ -10,8 +10,8 @@
 use crate::muxer::{create_muxer, Muxer};
 use crate::{ColorPrimaries, MatrixCoefficients, TransferCharacteristics};
 use clap::{App, AppSettings, Arg, ArgMatches, SubCommand, Shell};
-use rav1e::partition::BlockSize;
 use rav1e::*;
+use scan_fmt::scan_fmt;
```
Comment (kodabb) [inline src/bin/common.rs]: this one is used only in the cli, is this the right place?

Comment (lu-zero) [reply src/bin/common.rs]: yes, we could rename the filename so it is less misleading later :)

File: benches/bench.rs (hunk 1 above)
Comment (lu-zero) [reply benches/bench.rs]: I'll remove it now. Thank you for spotting.

**Comment by lu-zero** [comment]
> Massive improvement, thanks.
> 
> Context and ContextInner still both get exposed, can that be fixed?

Sure, I didn't touch that file to not collide with Tim's patch.

**Comment by lu-zero** [comment]
Now it looks like this:

Screenshot_2019-06-13 rav1e - Rust

**Comment by tdaede** [comment]
Perfect.

**Pull request merged**